<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tess Brodie" />


<title>IMC data analysis workflow</title>

<script src="site_libs/header-attrs-2.15/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">220707_workflowr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="0_prep_data.html">Data prep</a>
</li>
<li>
  <a href="1_QC_mask.html">QC mask</a>
</li>
<li>
  <a href="2_batch_effect_correction.html">Batch effect_corr</a>
</li>
<li>
  <a href="3_dim_red.html">Dim red</a>
</li>
<li>
  <a href="4_cell_annotation.html">Cell pheno</a>
</li>
<li>
  <a href="5_sc_vis.html">Single cell vis</a>
</li>
<li>
  <a href="6_img_vis.html">Image vis</a>
</li>
<li>
  <a href="47_spatial_vis.html">Spatial vis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">IMC data analysis workflow</h1>
<h4 class="author">Tess Brodie</h4>
<h4 class="date">University of Bern, DBMR, Visceral and Transplantation
Surgery lab</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>220707_workflowr/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220718code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220718)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220718code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220718)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
create Seurat object
</li>
<li>
FastMNN sample correction
</li>
<li>
harmony correction storage
</li>
<li>
integratoins by seurat
</li>
<li>
load-libs
</li>
<li>
load spe1
</li>
<li>
marker expression after correction
</li>
<li>
marker expression before correction
</li>
<li>
patient id visualization
</li>
<li>
save spe2
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
seurat pca
</li>
<li>
seurat UMAP
</li>
<li>
table batch effect and lost.var
</li>
<li>
UMAP before vs. after harmony correction
</li>
<li>
UMAP with corrected values
</li>
<li>
UMAP_after
</li>
<li>
UMAP_before
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>2_batch_effect_correction_cache</code> and re-run the analysis. To
have workflowr automatically delete the cache directory prior to
building the file, set <code>delete_cache = TRUE</code> when running
<code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/Users/IMC Platform
laptop/Users2/Tess/SteinbockCellSegmentation/220707/220707_workflowr
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongdb869f5">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> db869f5
</a>
</p>
</div>
<div id="strongRepositoryversionstrongdb869f5"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version db869f5.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/0_prep_data_cache/
    Ignored:    analysis/1_QC_mask_cache/
    Ignored:    analysis/2_batch_effect_correction_cache/
    Ignored:    analysis/3_dim_red_cache/
    Ignored:    analysis/4_cell_annotation_cache/
    Ignored:    analysis/5_sc_vis_cache/
    Ignored:    analysis/6_img_vis_cache/
    Ignored:    analysis/7_spatial_vis_cache/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  UMAP_correctedvalues.pdf
    Untracked:  Umap_celltype_overlay.pdf
    Untracked:  allchannels_histogram.pdf
    Untracked:  allchannels_multidimplot.pdf
    Untracked:  alldata_lsecs.rds
    Untracked:  data/Joseentraining/
    Untracked:  data/Macro.ijm.ijm.ijm
    Untracked:  data/cellpose png masks/
    Untracked:  data/cluster_annotation.png
    Untracked:  data/cluster_annotation.xlsx
    Untracked:  data/fiji composite images/
    Untracked:  data/images.csv
    Untracked:  data/images.rds
    Untracked:  data/img/
    Untracked:  data/intensities/
    Untracked:  data/masks.rds
    Untracked:  data/masks/
    Untracked:  data/neighbors/
    Untracked:  data/panel.csv
    Untracked:  data/raw/
    Untracked:  data/regionprops/
    Untracked:  data/sample_metadata.xlsx
    Untracked:  data/spe.rds
    Untracked:  gated_cells/
    Untracked:  gated_cells2/
    Untracked:  marker_umap_corrected.png
    Untracked:  output/spe0.rds
    Untracked:  output/spe1.rds
    Untracked:  output/spe2.rds
    Untracked:  output/spe3.rds
    Untracked:  output/spe4.rds

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/2_batch_effect_correction.Rmd</code>) and HTML
(<code>docs/2_batch_effect_correction.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
db869f5
</td>
<td>
tessbrodie
</td>
<td>
2022-08-30
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="batch-effect-correction" class="section level1">
<h1>Batch effect correction</h1>
<div
id="of-note-the-correction-approaches-presented-here-aim-at-removing-any-differences-between-samples.-this-will-also-remove-biological-differences-between-the-patientsindications.-nevertheless-integrating-cells-across-samples-can-facilitate-the-detection-of-cell-phenotypes-via-clustering."
class="section level3">
<h3>Of note: the correction approaches presented here aim at removing
any differences between samples. This will also remove biological
differences between the patients/indications. Nevertheless, integrating
cells across samples can facilitate the detection of cell phenotypes via
clustering.</h3>
<style type="text/css">
.scroll-100 {
  max-height: 100px;
  overflow-y: auto;
  background-color: inherit;
}
</style>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(batchelor)
library(scater)
library(cowplot)
library(dittoSeq)
library(viridis)
library(harmony)
library(Seurat)
library(SeuratObject)</code></pre>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<pre class="r"><code>spe &lt;- readRDS(file.path(&quot;output&quot;, &quot;spe1.rds&quot;))</code></pre>
</div>
<div id="fastmnn-method-to-correct-for-batch-effect"
class="section level3">
<h3>FastMNN method to correct for batch effect:</h3>
<pre class="r"><code>out &lt;- fastMNN(spe, batch = spe$patient_id,
               auto.merge = TRUE,
               assay.type = &quot;exprs&quot;)</code></pre>
<pre><code>Loading required package: SpatialExperiment</code></pre>
<pre><code>Warning in check_numbers(k = k, nu = nu, nv = nv, limit = min(dim(x)) - : more
singular values/vectors requested than available</code></pre>
<pre><code>Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =
TRUE, : You&#39;re computing too large a percentage of total singular values, use a
standard svd instead.</code></pre>
<pre class="r"><code># Transfer the correction results to the main spe object
reducedDim(spe, &quot;fastMNN&quot;) &lt;- reducedDim(out, &quot;corrected&quot;)</code></pre>
</div>
<div
id="qc-of-corrected-results-gives-batch-effect-and-lost-variance-values-if-combining-samples"
class="section level3">
<h3>QC of corrected results: gives batch effect and lost variance values
if combining samples</h3>
<pre class="r"><code>merge_info &lt;- metadata(out)$merge.info 

DataFrame(left = merge_info$left,
          right = merge_info$right,
          batch.size = merge_info$batch.size,
          max_lost_var = rowMax(merge_info$lost.var))</code></pre>
<pre class="scroll-100"><code>DataFrame with 5 rows and 4 columns
       left  right batch.size max_lost_var
     &lt;List&gt; &lt;List&gt;  &lt;numeric&gt;    &lt;numeric&gt;
1         6      5   0.357777    0.0451654
2       6,5      2   0.599930    0.0761382
3     6,5,2      1   0.860286    0.0442115
4 6,5,2,...      4   0.664935    0.1257983
5 6,5,2,...      3   0.946328    0.1132999</code></pre>
</div>
<div id="visualization-of-batch-corrections" class="section level3">
<h3>Visualization of batch corrections:</h3>
</div>
<div
id="recompute-the-umap-embedding-using-the-corrected-low-dimensional-coordinates-for-each-cell"
class="section level3">
<h3>recompute the UMAP embedding using the corrected low-dimensional
coordinates for each cell</h3>
<pre class="r"><code>spe &lt;- runUMAP(spe, dimred= &quot;fastMNN&quot;, name = &quot;UMAP_mnnCorrected&quot;) 

# visualize patient id 
p1 &lt;- dittoDimPlot(spe, var = &quot;patient_id&quot;, 
                   reduction.use = &quot;UMAP&quot;, size = 0.2) + 
    scale_color_manual(values = metadata(spe)$color_vectors$patient_id) +
    ggtitle(&quot;Patient ID on UMAP before correction&quot;)</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>p2 &lt;- dittoDimPlot(spe, var = &quot;patient_id&quot;, 
                   reduction.use = &quot;UMAP_mnnCorrected&quot;, size = 0.2) + 
    scale_color_manual(values = metadata(spe)$color_vectors$patient_id) +
    ggtitle(&quot;Patient ID on UMAP after correction&quot;)</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>#pdf(&quot;UMAP_correctedvalues.pdf&quot;, width = 20, height = 9)
plot_grid(p1, p2)</code></pre>
<p><img src="figure/2_batch_effect_correction.Rmd/UMAP with corrected values-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre class="scroll-100"><code>null device 
          1 </code></pre>
<pre class="r"><code>plot_grid(p1, p2)</code></pre>
</div>
<div
id="visualize-the-expression-of-selected-markers-across-all-cells-before-correction"
class="section level3">
<h3>Visualize the expression of selected markers across all cells before
correction</h3>
<pre class="r"><code>markers &lt;- c(&quot;E-cadherin&quot;, &quot;CD8a&quot;, &quot;CD4&quot;, &quot;CD3e&quot;, &quot;HNFa&quot;, &quot;Ki-67&quot;, &quot;CD19&quot;, &quot;a-SMA&quot;, &quot;CD11b&quot;, &quot;Ly6-G&quot;, &quot;Pan-actin&quot;, &quot;CD31&quot;, &quot;CD74&quot;, &quot;CD44&quot;, &quot;Arginase-1&quot;, &quot;F4_80&quot;, &quot;CD206&quot; )

# Before correction
plot_list &lt;- multi_dittoDimPlot(spe, var = markers, reduction.use = &quot;UMAP&quot;, 
                   assay = &quot;exprs&quot;, size = 0.2, list.out = TRUE) 


plot_list &lt;- lapply(plot_list, function(x) x + scale_color_viridis())</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>plot_grid(plotlist = plot_list) </code></pre>
<p><img src="figure/2_batch_effect_correction.Rmd/marker expression before correction-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="visualize-the-expression-of-selected-markers-across-all-cells-after-correction"
class="section level3">
<h3>Visualize the expression of selected markers across all cells after
correction</h3>
<pre class="r"><code># After correction
plot_list &lt;- multi_dittoDimPlot(spe, var = markers, reduction.use = &quot;UMAP_mnnCorrected&quot;, 
                   assay = &quot;exprs&quot;, size = 0.2, list.out = TRUE) 
plot_list &lt;- lapply(plot_list, function(x) x + scale_color_viridis())</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>plot_grid(plotlist = plot_list) </code></pre>
<p><img src="figure/2_batch_effect_correction.Rmd/marker expression after correction-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="harmony-correction" class="section level3">
<h3>Harmony correction:</h3>
</div>
<div
id="harmony-returns-the-corrected-low-dimensional-coordinates-for-each-cell"
class="section level3">
<h3>Harmony returns the corrected low-dimensional coordinates for each
cell</h3>
<pre class="r"><code>mat &lt;- t(assay(spe, &quot;exprs&quot;))

harmony_emb &lt;- HarmonyMatrix(mat, spe$patient_id, do_pca = TRUE, npcs = 15)</code></pre>
<pre><code>Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth =
TRUE, : You&#39;re computing too large a percentage of total singular values, use a
standard svd instead.</code></pre>
<pre><code>Harmony 1/10</code></pre>
<pre><code>Harmony 2/10</code></pre>
<pre><code>Harmony 3/10</code></pre>
<pre><code>Harmony 4/10</code></pre>
<pre><code>Harmony 5/10</code></pre>
<pre><code>Harmony 6/10</code></pre>
<pre><code>Harmony 7/10</code></pre>
<pre><code>Harmony 8/10</code></pre>
<pre><code>Harmony 9/10</code></pre>
<pre><code>Harmony 10/10</code></pre>
<pre><code>Harmony converged after 10 iterations</code></pre>
<pre class="r"><code>reducedDim(spe, &quot;UMAP_harmony&quot;) &lt;- harmony_emb</code></pre>
</div>
<div id="visualize-the-cells-in-low-dimensions-after-umap-embedding"
class="section level3">
<h3>Visualize the cells in low dimensions after UMAP embedding:</h3>
<pre class="r"><code># visualize patient id 
p1 &lt;- dittoDimPlot(spe, var = &quot;patient_id&quot;, 
                   reduction.use = &quot;UMAP&quot;, size = 0.2) + 
    scale_color_manual(values = metadata(spe)$color_vectors$patient_id) +
    ggtitle(&quot;Patient ID on UMAP before correction&quot;)</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>p2 &lt;- dittoDimPlot(spe, var = &quot;patient_id&quot;, 
                   reduction.use = &quot;UMAP_harmony&quot;, size = 0.2) + 
    scale_color_manual(values = metadata(spe)$color_vectors$patient_id) +
    ggtitle(&quot;Patient ID on UMAP after correction&quot;)</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>#pdf(&quot;UMAP_harmonycorrection.pdf&quot;, width = 20, height = 9)
plot_grid(p1, p2)</code></pre>
<p><img src="figure/2_batch_effect_correction.Rmd/UMAP before vs. after harmony correction-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
<pre class="r"><code>plot_grid(p1, p2)</code></pre>
</div>
<div
id="visualize-the-expression-of-selected-markers-across-all-cells-before-correction-1"
class="section level3">
<h3>Visualize the expression of selected markers across all cells before
correction</h3>
<pre class="r"><code># Before correction
plot_list &lt;- multi_dittoDimPlot(spe, var = markers, reduction.use = &quot;UMAP&quot;, 
                   assay = &quot;exprs&quot;, size = 0.2, list.out = TRUE) 
plot_list &lt;- lapply(plot_list, function(x) x + scale_color_viridis())</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>plot_grid(plotlist = plot_list) </code></pre>
<p><img src="figure/2_batch_effect_correction.Rmd/UMAP_before-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="visualize-the-expression-of-selected-markers-across-all-cells-after-correction-1"
class="section level3">
<h3>Visualize the expression of selected markers across all cells after
correction</h3>
<pre class="r"><code># After correction
plot_list &lt;- multi_dittoDimPlot(spe, var = markers, reduction.use = &quot;UMAP_harmony&quot;, 
                   assay = &quot;exprs&quot;, size = 0.2, list.out = TRUE) 
plot_list &lt;- lapply(plot_list, function(x) x + scale_color_viridis())</code></pre>
<pre><code>Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.
Scale for &#39;colour&#39; is already present. Adding another scale for &#39;colour&#39;,
which will replace the existing scale.</code></pre>
<pre class="r"><code>plot_grid(plotlist = plot_list) </code></pre>
<p><img src="figure/2_batch_effect_correction.Rmd/UMAP_after-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="seurat-correction" class="section level3">
<h3>Seurat correction</h3>
</div>
<div
id="seurat-performs-batch-correction-similarly-to-fastmnn-by-finding-mutual-nearest-neighbors-mnn-in-low-dimensional-space-before-correcting-the-expression-values-of-cells"
class="section level3">
<h3>Seurat performs batch correction similarly to fastMNN by finding
mutual nearest neighbors (MNN) in low dimensional space before
correcting the expression values of cells</h3>
</div>
<div id="integrate-markers-and-patient-ids-did-not-work-for-me"
class="section level2">
<h2>Integrate markers and patient Ids ### Did not work for me</h2>
<div id="pca-reduction-in-seurat-method" class="section level3">
<h3>PCA reduction in seurat method:</h3>
</div>
<div
id="recompute-the-umap-embeddings-based-on-seurat-integrated-results-and-visualize-the-embedding"
class="section level3">
<h3>Recompute the UMAP embeddings based on Seurat integrated results and
visualize the embedding</h3>
</div>
<div id="visualize-patient-id" class="section level3">
<h3>Visualize patient id</h3>
</div>
<div
id="visualize-the-expression-of-selected-markers-across-all-cells-before-correction-2"
class="section level3">
<h3>Visualize the expression of selected markers across all cells before
correction</h3>
</div>
<div
id="visualize-the-expression-of-selected-markers-across-all-cells-after-correction-2"
class="section level3">
<h3>Visualize the expression of selected markers across all cells after
correction</h3>
</div>
<div id="save-data" class="section level3">
<h3>Save data</h3>
<pre class="r"><code>saveRDS(spe, file.path(&quot;output&quot;, &quot;spe2.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Switzerland.utf8  LC_CTYPE=English_Switzerland.utf8   
[3] LC_MONETARY=English_Switzerland.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Switzerland.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] SpatialExperiment_1.6.0     sp_1.5-0                   
 [3] SeuratObject_4.1.0          Seurat_4.1.1               
 [5] harmony_0.1.0               Rcpp_1.0.9                 
 [7] viridis_0.6.2               viridisLite_0.4.0          
 [9] dittoSeq_1.8.1              cowplot_1.1.1              
[11] scater_1.24.0               ggplot2_3.3.6              
[13] scuttle_1.6.2               batchelor_1.12.3           
[15] SingleCellExperiment_1.18.0 SummarizedExperiment_1.26.1
[17] Biobase_2.56.0              GenomicRanges_1.48.0       
[19] GenomeInfoDb_1.32.2         IRanges_2.30.0             
[21] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[23] MatrixGenerics_1.8.1        matrixStats_0.62.0         
[25] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                R.utils_2.12.0           
  [3] reticulate_1.25           tidyselect_1.1.2         
  [5] htmlwidgets_1.5.4         grid_4.2.1               
  [7] BiocParallel_1.30.3       Rtsne_0.16               
  [9] DropletUtils_1.16.0       munsell_0.5.0            
 [11] ScaledMatrix_1.4.0        codetools_0.2-18         
 [13] ica_1.0-3                 future_1.26.1            
 [15] miniUI_0.1.1.1            withr_2.5.0              
 [17] spatstat.random_2.2-0     colorspace_2.0-3         
 [19] progressr_0.10.1          highr_0.9                
 [21] knitr_1.39                rstudioapi_0.13          
 [23] ROCR_1.0-11               tensor_1.5               
 [25] listenv_0.8.0             labeling_0.4.2           
 [27] git2r_0.30.1              GenomeInfoDbData_1.2.8   
 [29] polyclip_1.10-0           farver_2.1.1             
 [31] pheatmap_1.0.12           rhdf5_2.40.0             
 [33] rprojroot_2.0.3           parallelly_1.32.0        
 [35] vctrs_0.4.1               generics_0.1.3           
 [37] xfun_0.31                 R6_2.5.1                 
 [39] ggbeeswarm_0.6.0          rsvd_1.0.5               
 [41] locfit_1.5-9.6            bitops_1.0-7             
 [43] rhdf5filters_1.8.0        spatstat.utils_2.3-1     
 [45] cachem_1.0.6              DelayedArray_0.22.0      
 [47] assertthat_0.2.1          promises_1.2.0.1         
 [49] scales_1.2.0              rgeos_0.5-9              
 [51] beeswarm_0.4.0            gtable_0.3.0             
 [53] beachmat_2.12.0           globals_0.15.1           
 [55] processx_3.7.0            goftest_1.2-3            
 [57] rlang_1.0.4               splines_4.2.1            
 [59] lazyeval_0.2.2            spatstat.geom_2.4-0      
 [61] yaml_2.3.5                reshape2_1.4.4           
 [63] abind_1.4-5               httpuv_1.6.5             
 [65] tools_4.2.1               ellipsis_0.3.2           
 [67] spatstat.core_2.4-4       jquerylib_0.1.4          
 [69] RColorBrewer_1.1-3        ggridges_0.5.3           
 [71] plyr_1.8.7                sparseMatrixStats_1.8.0  
 [73] zlibbioc_1.42.0           purrr_0.3.4              
 [75] RCurl_1.98-1.7            ps_1.7.1                 
 [77] rpart_4.1.16              deldir_1.0-6             
 [79] pbapply_1.5-0             zoo_1.8-10               
 [81] ggrepel_0.9.1             cluster_2.1.3            
 [83] fs_1.5.2                  magrittr_2.0.3           
 [85] RSpectra_0.16-1           magick_2.7.3             
 [87] data.table_1.14.2         scattermore_0.8          
 [89] ResidualMatrix_1.6.0      lmtest_0.9-40            
 [91] RANN_2.6.1                whisker_0.4              
 [93] fitdistrplus_1.1-8        patchwork_1.1.1          
 [95] mime_0.12                 evaluate_0.15            
 [97] xtable_1.8-4              gridExtra_2.3            
 [99] compiler_4.2.1            tibble_3.1.7             
[101] KernSmooth_2.23-20        R.oo_1.25.0              
[103] htmltools_0.5.3           mgcv_1.8-40              
[105] later_1.3.0               tidyr_1.2.0              
[107] DBI_1.1.3                 MASS_7.3-58              
[109] Matrix_1.4-1              cli_3.3.0                
[111] R.methodsS3_1.8.2         parallel_4.2.1           
[113] igraph_1.3.4              pkgconfig_2.0.3          
[115] getPass_0.2-2             plotly_4.10.0            
[117] spatstat.sparse_2.1-1     vipor_0.4.5              
[119] bslib_0.4.0               dqrng_0.3.0              
[121] XVector_0.36.0            stringr_1.4.0            
[123] callr_3.7.1               digest_0.6.29            
[125] sctransform_0.3.3         RcppAnnoy_0.0.19         
[127] spatstat.data_2.2-0       rmarkdown_2.14           
[129] leiden_0.4.2              edgeR_3.38.1             
[131] uwot_0.1.11               DelayedMatrixStats_1.18.0
[133] shiny_1.7.2               rjson_0.2.21             
[135] lifecycle_1.0.1           nlme_3.1-158             
[137] jsonlite_1.8.0            Rhdf5lib_1.18.2          
[139] BiocNeighbors_1.14.0      limma_3.52.2             
[141] fansi_1.0.3               pillar_1.8.0             
[143] lattice_0.20-45           fastmap_1.1.0            
[145] httr_1.4.3                survival_3.3-1           
[147] glue_1.6.2                png_0.1-7                
[149] stringi_1.7.8             sass_0.4.2               
[151] HDF5Array_1.24.1          BiocSingular_1.12.0      
[153] dplyr_1.0.9               irlba_2.3.5              
[155] future.apply_1.9.0       </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
