<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tess Brodie" />


<title>IMC data analysis workflow</title>

<script src="site_libs/header-attrs-2.15/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">220707_workflowr</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="0_prep_data.html">Data prep</a>
</li>
<li>
  <a href="1_QC_mask.html">QC mask</a>
</li>
<li>
  <a href="2_batch_effect_correction.html">Batch effect_corr</a>
</li>
<li>
  <a href="3_dim_red.html">Dim red</a>
</li>
<li>
  <a href="4_cell_annotation.html">Cell pheno</a>
</li>
<li>
  <a href="5_sc_vis.html">Single cell vis</a>
</li>
<li>
  <a href="6_img_vis.html">Image vis</a>
</li>
<li>
  <a href="47_spatial_vis.html">Spatial vis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">IMC data analysis workflow</h1>
<h4 class="author">Tess Brodie</h4>
<h4 class="date">University of Bern, DBMR, Visceral and Transplantation
Surgery lab</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>220707_workflowr/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220718code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220718)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220718code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220718)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
approach comparison cell integration
</li>
<li>
approach comparison
</li>
<li>
bluster clustering
</li>
<li>
bluster corrected
</li>
<li>
CATALYST phenograph
</li>
<li>
create heatmap with integrated phenoannoy
</li>
<li>
create heatmap with phenoannoy data
</li>
<li>
cytomapperShiny
</li>
<li>
heatmap bluster corrected
</li>
<li>
heatmap louvain integrated
</li>
<li>
heatmap louvain
</li>
<li>
heatmap metaclusters
</li>
<li>
integrated cell phenoannoy
</li>
<li>
load-libs
</li>
<li>
load data
</li>
<li>
louvain integrated phenograph
</li>
<li>
louvain phenograph
</li>
<li>
metaclustering
</li>
<li>
Rphenoannoy
</li>
<li>
save spe3
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
shared nearest neigbhours graph
</li>
<li>
unnamed-chunk-1
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>3_dim_red_cache</code> and re-run the analysis. To have workflowr
automatically delete the cache directory prior to building the file, set
<code>delete_cache = TRUE</code> when running <code>wflow_build()</code>
or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/Users/IMC Platform
laptop/Users2/Tess/SteinbockCellSegmentation/220707/220707_workflowr
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongdb869f5">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> db869f5
</a>
</p>
</div>
<div id="strongRepositoryversionstrongdb869f5"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version db869f5.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/0_prep_data_cache/
    Ignored:    analysis/1_QC_mask_cache/
    Ignored:    analysis/2_batch_effect_correction_cache/
    Ignored:    analysis/3_dim_red_cache/
    Ignored:    analysis/4_cell_annotation_cache/
    Ignored:    analysis/5_sc_vis_cache/
    Ignored:    analysis/6_img_vis_cache/
    Ignored:    analysis/7_spatial_vis_cache/

Untracked files:
    Untracked:  Rplot.png
    Untracked:  UMAP_correctedvalues.pdf
    Untracked:  Umap_celltype_overlay.pdf
    Untracked:  allchannels_histogram.pdf
    Untracked:  allchannels_multidimplot.pdf
    Untracked:  alldata_lsecs.rds
    Untracked:  data/Joseentraining/
    Untracked:  data/Macro.ijm.ijm.ijm
    Untracked:  data/cellpose png masks/
    Untracked:  data/cluster_annotation.png
    Untracked:  data/cluster_annotation.xlsx
    Untracked:  data/fiji composite images/
    Untracked:  data/images.csv
    Untracked:  data/images.rds
    Untracked:  data/img/
    Untracked:  data/intensities/
    Untracked:  data/masks.rds
    Untracked:  data/masks/
    Untracked:  data/neighbors/
    Untracked:  data/panel.csv
    Untracked:  data/raw/
    Untracked:  data/regionprops/
    Untracked:  data/sample_metadata.xlsx
    Untracked:  data/spe.rds
    Untracked:  gated_cells/
    Untracked:  gated_cells2/
    Untracked:  marker_umap_corrected.png
    Untracked:  output/spe0.rds
    Untracked:  output/spe1.rds
    Untracked:  output/spe2.rds
    Untracked:  output/spe3.rds
    Untracked:  output/spe4.rds

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/3_dim_red.Rmd</code>) and HTML
(<code>docs/3_dim_red.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
db869f5
</td>
<td>
tessbrodie
</td>
<td>
2022-08-30
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="dimensionality-reduction-and-cell-classification"
class="section level1">
<h1>Dimensionality reduction and cell classification</h1>
<style type="text/css">
.scroll-100 {
  max-height: 100px;
  overflow-y: auto;
  background-color: inherit;
}
</style>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(SpatialExperiment)
library(Rphenograph)
library(igraph)
library(dittoSeq)
library(viridis)
library(bluster)
library(BiocParallel)
library(ggplot2)
library(scran)
library(CATALYST)
library(kohonen)
library(ConsensusClusterPlus)
library(patchwork)
library(pheatmap)
library(gridExtra)
library(cytomapper)
library(Rphenoannoy)
library(SingleCellExperiment)
library(caret)
library(ggridges)</code></pre>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<pre class="r"><code>spe &lt;- readRDS(file.path(&quot;output&quot;, &quot;spe2.rds&quot;))

# Sample cells
cur_cells &lt;- sample(seq_len(ncol(spe)), 2000)</code></pre>
</div>
<div id="dimensionality-reduction-rphenoannoy" class="section level3">
<h3>Dimensionality reduction: Rphenoannoy</h3>
<pre class="r"><code>mat &lt;- t(assay(spe, &quot;exprs&quot;))

out &lt;- Rphenoannoy(mat, k = 10)</code></pre>
<pre><code>Run Rphenograph starts:
  -Input data of 15521 rows and 23 columns
  -k is set to 10</code></pre>
<pre><code>  Finding nearest neighbors...DONE ~ 2.78 s
  Compute jaccard coefficient between nearest-neighbor sets...</code></pre>
<pre><code>Presorting knn...</code></pre>
<pre><code>presorting DONE ~ 0.58 s
  Start jaccard
DONE ~ 0.01 s
  Build undirected graph from the weighted links...DONE ~ 0.06 s
  Run louvain clustering on the graph ...DONE ~ 0.23 s</code></pre>
<pre><code>Run Rphenograph DONE, totally takes 3.08s.</code></pre>
<pre><code>  Return a community class
  -Modularity value: 0.874768 
  -Number of clusters: 33</code></pre>
<pre class="r"><code>clusters &lt;- factor(membership(out[[2]]))

spe$pg_clusters &lt;- clusters

dittoDimPlot(spe, var = &quot;pg_clusters&quot;, 
             reduction.use = &quot;UMAP&quot;, size = 0.2,
             do.label = TRUE) +
    ggtitle(&quot;Phenoannoy clusters expression on UMAP&quot;)</code></pre>
<p><img src="figure/3_dim_red.Rmd/Rphenoannoy-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="heatmap-with-phenoannoy-clusters" class="section level1">
<h1>Heatmap with phenoannoy clusters:</h1>
<pre class="r"><code>dittoHeatmap(spe[, cur_cells], 
             genes = rownames(spe),
             assay = &quot;exprs&quot;, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), 
             annot.by = c(&quot;pg_clusters&quot;, &quot;patient_id&quot;),
             annot.colors = c(dittoColors(1)[1:length(unique(spe$pg_clusters))],
                              metadata(spe)$color_vectors$patient_id))</code></pre>
<p><img src="figure/3_dim_red.Rmd/create heatmap with phenoannoy data-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="create-a-phenoannoy-with-fastmnn-cell-integration-data"
class="section level3">
<h3>Create a phenoannoy with FastMNN cell integration data</h3>
<pre class="r"><code>set.seed(322)

mat &lt;- reducedDim(spe, &quot;fastMNN&quot;)

out &lt;- Rphenoannoy(mat, k = 10)</code></pre>
<pre><code>Run Rphenograph starts:
  -Input data of 15521 rows and 22 columns
  -k is set to 10</code></pre>
<pre><code>  Finding nearest neighbors...DONE ~ 2.79 s
  Compute jaccard coefficient between nearest-neighbor sets...</code></pre>
<pre><code>Presorting knn...</code></pre>
<pre><code>presorting DONE ~ 0.58 s
  Start jaccard
DONE ~ 0 s
  Build undirected graph from the weighted links...DONE ~ 0.05 s
  Run louvain clustering on the graph ...DONE ~ 0.19 s</code></pre>
<pre><code>Run Rphenograph DONE, totally takes 3.03s.</code></pre>
<pre><code>  Return a community class
  -Modularity value: 0.8722263 
  -Number of clusters: 34</code></pre>
<pre class="r"><code>clusters &lt;- factor(membership(out[[2]]))

spe$pg_clusters_corrected &lt;- clusters

dittoDimPlot(spe, var = &quot;pg_clusters_corrected&quot;, 
             reduction.use = &quot;UMAP_mnnCorrected&quot;, size = 0.2,
             do.label = TRUE) +
    ggtitle(&quot;Phenoannoy clusters expression on UMAP, integrated cells&quot;)</code></pre>
<p><img src="figure/3_dim_red.Rmd/integrated cell phenoannoy-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-with-fastmnn-cell-integrated-phenoannoy"
class="section level3">
<h3>Heatmap with FastMNN cell integrated phenoannoy</h3>
<pre class="r"><code>dittoHeatmap(spe[,cur_cells], 
             genes = rownames(spe),
             assay = &quot;exprs&quot;, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), 
             annot.by = c(&quot;pg_clusters_corrected&quot;, &quot;patient_id&quot;),
             annot.colors = c(dittoColors(1)[1:length(unique(spe$pg_clusters_corrected))],
                              metadata(spe)$color_vectors$patient_id))</code></pre>
<p><img src="figure/3_dim_red.Rmd/create heatmap with integrated phenoannoy-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="shared-nearest-neighbour-graph-this-takes-a-while-to-run"
class="section level3">
<h3>Shared nearest neighbour graph: This takes a while to run</h3>
</div>
<div
id="best-value-for-average-silhouette-width-is-louvain-rank-cluster-k20"
class="section level3">
<h3>Best value for average silhouette width is louvain rank cluster
k=20</h3>
</div>
<div
id="phenograph-with-louvain-community-detection-and-rank-based-weighting-of-edges-k20"
class="section level3">
<h3>Phenograph with louvain community detection and rank-based weighting
of edges (k=20)</h3>
<pre class="r"><code>clusters &lt;- clusterCells(spe, 
                         assay.type = &quot;exprs&quot;, 
                         BLUSPARAM = NNGraphParam(k = 20, 
                                                  cluster.fun = &quot;louvain&quot;,
                                                  type = &quot;rank&quot;,
                                                  BPPARAM = bpparam()))

spe$nn_clusters &lt;- clusters

dittoDimPlot(spe, var = &quot;nn_clusters&quot;, 
             reduction.use = &quot;UMAP&quot;, size = 0.2,
             do.label = TRUE) +
    ggtitle(&quot;SNN clusters expression on UMAP&quot;)</code></pre>
<p><img src="figure/3_dim_red.Rmd/louvain phenograph-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-louvain---rank---k20" class="section level3">
<h3>Heatmap louvain - rank - k=20</h3>
<pre class="r"><code>dittoHeatmap(spe[,cur_cells], 
             genes = rownames(spe),
             assay = &quot;exprs&quot;, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), 
             annot.by = c(&quot;nn_clusters&quot;, &quot;patient_id&quot;),
             annot.colors = c(dittoColors(1)[1:length(unique(spe$nn_clusters))],
                              metadata(spe)$color_vectors$patient_id))</code></pre>
<p><img src="figure/3_dim_red.Rmd/heatmap louvain-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="phenograph-louvain-with-integrated-cells"
class="section level3">
<h3>Phenograph louvain with integrated cells</h3>
<pre class="r"><code>clusters &lt;- clusterCells(spe, 
                         use.dimred = &quot;fastMNN&quot;, 
                         BLUSPARAM = NNGraphParam(k = 20, 
                                                  cluster.fun = &quot;louvain&quot;,
                                                  type = &quot;rank&quot;,
                                                  BPPARAM = bpparam()))

spe$nn_clusters_corrected &lt;- clusters

dittoDimPlot(spe, var = &quot;nn_clusters_corrected&quot;, 
             reduction.use = &quot;UMAP_mnnCorrected&quot;, size = 0.2,
             do.label = TRUE) +
    ggtitle(&quot;SNN clusters expression on UMAP, integrated cells&quot;)</code></pre>
<p><img src="figure/3_dim_red.Rmd/louvain integrated phenograph-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-louvain-integrated-phenograph" class="section level3">
<h3>Heatmap louvain integrated phenograph</h3>
<pre class="r"><code>dittoHeatmap(spe[,cur_cells], 
             genes = rownames(spe),
             assay = &quot;exprs&quot;, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), 
             annot.by = c(&quot;nn_clusters_corrected&quot;, &quot;patient_id&quot;),
             annot.colors = c(dittoColors(1)[1:length(unique(spe$nn_clusters_corrected))],
                              metadata(spe)$color_vectors$patient_id))</code></pre>
<p><img src="figure/3_dim_red.Rmd/heatmap louvain integrated-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="flowsomcatalyst-phenographs" class="section level3">
<h3>flowSOM/Catalyst phenographs</h3>
<pre class="r"><code># Run FlowSOM and ConsensusClusterPlus clustering
spe &lt;- CATALYST::cluster(spe, 
               features = rownames(spe),
               maxK = 30)</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre class="r"><code># Assess cluster stability
delta_area(spe)</code></pre>
<p><img src="figure/3_dim_red.Rmd/CATALYST phenograph-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-clusters-with-meta12" class="section level3">
<h3>Cell clusters with meta12</h3>
<pre class="r"><code>spe$som_clusters &lt;- cluster_ids(spe, &quot;meta12&quot;)

dittoDimPlot(spe, var = &quot;som_clusters&quot;, 
             reduction.use = &quot;UMAP&quot;, size = 0.2,
             do.label = TRUE) +
    ggtitle(&quot;SOM clusters expression on UMAP&quot;)</code></pre>
<p><img src="figure/3_dim_red.Rmd/metaclustering-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap" class="section level3">
<h3>Heatmap</h3>
<pre class="r"><code>dittoHeatmap(spe[,cur_cells], 
             genes = rownames(spe),
             assay = &quot;exprs&quot;, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), 
             annot.by = c(&quot;som_clusters&quot;, &quot;patient_id&quot;),
             annot.colors = c(dittoColors(1)[1:length(unique(spe$som_clusters))],
                              metadata(spe)$color_vectors$patient_id))</code></pre>
<p><img src="figure/3_dim_red.Rmd/heatmap metaclusters-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="clustering-with-bluster-package" class="section level3">
<h3>Clustering with bluster package</h3>
<pre class="r"><code># Select integrated cells
mat &lt;- reducedDim(spe, &quot;fastMNN&quot;)

# Perform SOM clustering
som.out &lt;- clusterRows(mat, SomParam(100), full = TRUE)

# Cluster the 100 SOM codes into larger clusters
ccp &lt;- ConsensusClusterPlus(t(som.out$objects$som$codes[[1]]),
                            maxK = 30,
                            reps = 100, 
                            distance = &quot;euclidean&quot;, 
                            seed = 220410, 
                            plot = NULL)</code></pre>
<pre><code>end fraction</code></pre>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-10.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-11.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-12.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-13.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-14.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-15.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-16.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-17.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-18.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-19.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-20.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-21.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-22.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-23.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-24.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-25.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-26.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-27.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-28.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>clustered</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-29.png" width="672" style="display: block; margin: auto;" /><img src="figure/3_dim_red.Rmd/bluster clustering-30.png" width="672" style="display: block; margin: auto;" /><img src="figure/3_dim_red.Rmd/bluster clustering-31.png" width="672" style="display: block; margin: auto;" /><img src="figure/3_dim_red.Rmd/bluster clustering-32.png" width="672" style="display: block; margin: auto;" /><img src="figure/3_dim_red.Rmd/bluster clustering-33.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CATALYST:::.plot_delta_area(ccp)</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster clustering-34.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="link-consensusclusterplus-clusters-with-som-codes-and-save-in-object"
class="section level3">
<h3>Link ConsensusClusterPlus clusters with SOM codes and save in
object</h3>
<pre class="r"><code>som.cluster &lt;- ccp[[16]][[&quot;consensusClass&quot;]][som.out$clusters]
spe$som_clusters_corrected &lt;- as.factor(som.cluster)

dittoDimPlot(spe, var = &quot;som_clusters_corrected&quot;, 
             reduction.use = &quot;UMAP_mnnCorrected&quot;, size = 0.2,
             do.label = TRUE) +
    ggtitle(&quot;Phenograph clusters expression on UMAP, integrated cells&quot;)</code></pre>
<p><img src="figure/3_dim_red.Rmd/bluster corrected-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-with-som" class="section level3">
<h3>Heatmap with SOM</h3>
<pre class="r"><code>dittoHeatmap(spe[,cur_cells], 
             genes = rownames(spe),
             assay = &quot;exprs&quot;, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), 
             annot.by = c(&quot;som_clusters_corrected&quot;, &quot;patient_id&quot;),
             annot.colors = c(dittoColors(1)[1:length(unique(spe$som_clusters_corrected))],
                              metadata(spe)$color_vectors$patient_id))</code></pre>
<p><img src="figure/3_dim_red.Rmd/heatmap bluster corrected-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-between-clustering-approaches"
class="section level3">
<h3>Comparison between clustering approaches</h3>
<pre class="r"><code>tab1 &lt;- table(paste(&quot;Rphenograph&quot;, spe$pg_clusters), 
              paste(&quot;SNN&quot;, spe$nn_clusters))
tab2 &lt;- table(paste(&quot;Rphenograph&quot;, spe$pg_clusters), 
              paste(&quot;SOM&quot;, spe$som_clusters))
tab3 &lt;- table(paste(&quot;SNN&quot;, spe$nn_clusters), 
              paste(&quot;SOM&quot;, spe$som_clusters))

ph1 &lt;- pheatmap(log10(tab1 + 10), color = viridis(100))[[4]]</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ph2 &lt;- pheatmap(log10(tab2 + 10), color = viridis(100))[[4]]</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ph3 &lt;- pheatmap(log10(tab3 + 10), color = viridis(100))[[4]]</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>grid.arrange(ph1, ph2, ph3, nrow = 1, ncol = 3)</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="comparison-between-clustering-approaches-integrated-cells"
class="section level3">
<h3>Comparison between clustering approaches: integrated cells</h3>
<pre class="r"><code>tab1 &lt;- table(paste(&quot;Rphenograph&quot;, spe$pg_clusters_corrected), 
              paste(&quot;SNN&quot;, spe$nn_clusters_corrected))
tab2 &lt;- table(paste(&quot;Rphenograph&quot;, spe$pg_clusters_corrected), 
              paste(&quot;SOM&quot;, spe$som_clusters_corrected))
tab3 &lt;- table(paste(&quot;SNN&quot;, spe$nn_clusters_corrected), 
              paste(&quot;SOM&quot;, spe$som_clusters_corrected))

ph1 &lt;- pheatmap(log10(tab1 + 10), color = viridis(100))[[4]]</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison cell integration-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ph2 &lt;- pheatmap(log10(tab2 + 10), color = viridis(100))[[4]]</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison cell integration-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ph3 &lt;- pheatmap(log10(tab3 + 10), color = viridis(100))[[4]]</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison cell integration-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>grid.arrange(ph1, ph2, ph3, nrow = 1, ncol = 3)</code></pre>
<p><img src="figure/3_dim_red.Rmd/approach comparison cell integration-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="save-data" class="section level3">
<h3>Save data</h3>
<pre class="r"><code>saveRDS(spe, file.path(&quot;output&quot;, &quot;spe3.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Switzerland.utf8  LC_CTYPE=English_Switzerland.utf8   
[3] LC_MONETARY=English_Switzerland.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Switzerland.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ggridges_0.5.3              caret_6.0-92               
 [3] lattice_0.20-45             Rphenoannoy_0.1.0          
 [5] Matrix_1.4-1                cytomapper_1.8.0           
 [7] EBImage_4.38.0              gridExtra_2.3              
 [9] pheatmap_1.0.12             patchwork_1.1.1            
[11] ConsensusClusterPlus_1.60.0 kohonen_3.0.11             
[13] CATALYST_1.20.1             scran_1.24.0               
[15] scuttle_1.6.2               BiocParallel_1.30.3        
[17] bluster_1.6.0               viridis_0.6.2              
[19] viridisLite_0.4.0           dittoSeq_1.8.1             
[21] Rphenograph_0.99.1          igraph_1.3.4               
[23] ggplot2_3.3.6               SpatialExperiment_1.6.0    
[25] SingleCellExperiment_1.18.0 SummarizedExperiment_1.26.1
[27] Biobase_2.56.0              GenomicRanges_1.48.0       
[29] GenomeInfoDb_1.32.2         IRanges_2.30.0             
[31] S4Vectors_0.34.0            BiocGenerics_0.42.0        
[33] MatrixGenerics_1.8.1        matrixStats_0.62.0         
[35] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] scattermore_0.8           flowWorkspace_4.8.0      
  [3] ModelMetrics_1.2.2.2      R.methodsS3_1.8.2        
  [5] tidyr_1.2.0               knitr_1.39               
  [7] irlba_2.3.5               multcomp_1.4-19          
  [9] DelayedArray_0.22.0       R.utils_2.12.0           
 [11] rpart_4.1.16              data.table_1.14.2        
 [13] hardhat_1.2.0             RCurl_1.98-1.7           
 [15] doParallel_1.0.17         generics_0.1.3           
 [17] flowCore_2.8.0            ScaledMatrix_1.4.0       
 [19] terra_1.5-34              callr_3.7.1              
 [21] cowplot_1.1.1             TH.data_1.1-1            
 [23] RANN_2.6.1                future_1.26.1            
 [25] ggpointdensity_0.1.0      lubridate_1.8.0          
 [27] xml2_1.3.3                httpuv_1.6.5             
 [29] assertthat_0.2.1          gower_1.0.0              
 [31] xfun_0.31                 jquerylib_0.1.4          
 [33] evaluate_0.15             promises_1.2.0.1         
 [35] fansi_1.0.3               htmlwidgets_1.5.4        
 [37] Rgraphviz_2.40.0          DBI_1.1.3                
 [39] purrr_0.3.4               ellipsis_0.3.2           
 [41] dplyr_1.0.9               ggcyto_1.24.1            
 [43] ggnewscale_0.4.7          ggpubr_0.4.0             
 [45] backports_1.4.1           cytolib_2.8.0            
 [47] svgPanZoom_0.3.4          RcppParallel_5.1.5       
 [49] deldir_1.0-6              sparseMatrixStats_1.8.0  
 [51] vctrs_0.4.1               abind_1.4-5              
 [53] cachem_1.0.6              withr_2.5.0              
 [55] ggforce_0.3.3             aws.signature_0.6.0      
 [57] svglite_2.1.0             cluster_2.1.3            
 [59] crayon_1.5.1              drc_3.0-1                
 [61] recipes_1.0.1             edgeR_3.38.1             
 [63] pkgconfig_2.0.3           tweenr_1.0.2             
 [65] nlme_3.1-158              vipor_0.4.5              
 [67] nnet_7.3-17               globals_0.15.1           
 [69] rlang_1.0.4               lifecycle_1.0.1          
 [71] sandwich_3.0-2            rsvd_1.0.5               
 [73] rprojroot_2.0.3           polyclip_1.10-0          
 [75] tiff_0.1-11               graph_1.74.0             
 [77] raster_3.5-21             carData_3.0-5            
 [79] Rhdf5lib_1.18.2           zoo_1.8-10               
 [81] base64enc_0.1-3           beeswarm_0.4.0           
 [83] whisker_0.4               GlobalOptions_0.1.2      
 [85] processx_3.7.0            png_0.1-7                
 [87] rjson_0.2.21              shinydashboard_0.7.2     
 [89] bitops_1.0-7              getPass_0.2-2            
 [91] R.oo_1.25.0               pROC_1.18.0              
 [93] rhdf5filters_1.8.0        DelayedMatrixStats_1.18.0
 [95] shape_1.4.6               stringr_1.4.0            
 [97] parallelly_1.32.0         jpeg_0.1-9               
 [99] rstatix_0.7.0             ggsignif_0.6.3           
[101] aws.s3_0.3.21             beachmat_2.12.0          
[103] scales_1.2.0              magrittr_2.0.3           
[105] plyr_1.8.7                hexbin_1.28.2            
[107] zlibbioc_1.42.0           compiler_4.2.1           
[109] dqrng_0.3.0               RColorBrewer_1.1-3       
[111] plotrix_3.8-2             clue_0.3-61              
[113] cli_3.3.0                 XVector_0.36.0           
[115] listenv_0.8.0             ncdfFlow_2.42.1          
[117] ps_1.7.1                  FlowSOM_2.4.0            
[119] MASS_7.3-58               tidyselect_1.1.2         
[121] stringi_1.7.8             RProtoBufLib_2.8.0       
[123] highr_0.9                 yaml_2.3.5               
[125] BiocSingular_1.12.0       locfit_1.5-9.6           
[127] latticeExtra_0.6-30       ggrepel_0.9.1            
[129] grid_4.2.1                sass_0.4.2               
[131] tools_4.2.1               future.apply_1.9.0       
[133] parallel_4.2.1            CytoML_2.8.0             
[135] circlize_0.4.15           rstudioapi_0.13          
[137] foreach_1.5.2             git2r_0.30.1             
[139] metapod_1.4.0             prodlim_2019.11.13       
[141] farver_2.1.1              Rtsne_0.16               
[143] DropletUtils_1.16.0       digest_0.6.29            
[145] lava_1.6.10               shiny_1.7.2              
[147] Rcpp_1.0.9                car_3.1-0                
[149] broom_1.0.0               later_1.3.0              
[151] httr_1.4.3                ComplexHeatmap_2.12.0    
[153] colorspace_2.0-3          XML_3.99-0.10            
[155] fs_1.5.2                  splines_4.2.1            
[157] statmod_1.4.36            RBGL_1.72.0              
[159] scater_1.24.0             sp_1.5-0                 
[161] systemfonts_1.0.4         xtable_1.8-4             
[163] jsonlite_1.8.0            timeDate_4021.104        
[165] ipred_0.9-13              R6_2.5.1                 
[167] mime_0.12                 pillar_1.8.0             
[169] htmltools_0.5.3           nnls_1.4                 
[171] glue_1.6.2                fastmap_1.1.0            
[173] fftwtools_0.9-11          BiocNeighbors_1.14.0     
[175] class_7.3-20              codetools_0.2-18         
[177] mvtnorm_1.1-3             utf8_1.2.2               
[179] bslib_0.4.0               tibble_3.1.7             
[181] curl_4.3.2                ggbeeswarm_0.6.0         
[183] colorRamps_2.3.1          gtools_3.9.3             
[185] magick_2.7.3              interp_1.1-3             
[187] survival_3.3-1            limma_3.52.2             
[189] rmarkdown_2.14            munsell_0.5.0            
[191] GetoptLong_1.0.5          rhdf5_2.40.0             
[193] GenomeInfoDbData_1.2.8    iterators_1.0.14         
[195] HDF5Array_1.24.1          reshape2_1.4.4           
[197] gtable_0.3.0             </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
